packages/edit-site/src/store/actions.js

```diff
@@ -135,7 +135,11 @@ export function setHomeTemplateId( homeTemplateId ) {
  */
 export function* setPage( page ) {
 	if ( ! page.path && page.context?.postId ) {
-		page.path = `?p=${ page.context.postId }`;
+		if ( page.context?.postType === 'page' ) {
```

Comment:
The problem lies to the fact that previous path didn't start with the `/`.

`page.path = `/?p=${ page.context.postId }` should be fine.

Reply:
That will not always work, as WP can be installed in the subdirectory of a server, like `example.com/wordpress-installed-here/`.

We would therefore need to qualify the URL against the home URL of the site, which it looks like can be had with `select( 'core' ).getSite().url`.